cpp/modules/deck.gl/core/src/lib/deck.cc

```diff
@@ -104,6 +108,10 @@ void Deck::setProps(std::shared_ptr<Deck::Props> props) {
   this->animationLoop->setSize({props->width, props->height});
 }
 
+void Deck::setProps(std::shared_ptr<Deck::Props> props, probegl::Error* error) {
```

Comment:
Maybe list this one first, as part of the "protected" API of constructor + some general methods like setProps and run. 

Then a section with potentially "non-protected APIs"? We don't need to protect every convenience function at this stage. The user can see which functions take an Error and stick to those.